src/Microsoft.ML.Data/Transforms/ExtensionsCatalog.cs

```diff
@@ -107,8 +107,28 @@ public static ColumnSelectingEstimator DropColumns(this TransformsCatalog catalo
         /// </example>
         public static ColumnSelectingEstimator SelectColumns(this TransformsCatalog catalog,
             string[] keepColumns,
-            bool keepHidden = ColumnSelectingTransformer.Defaults.KeepHidden)
+            bool keepHidden)
             => new ColumnSelectingEstimator(CatalogUtils.GetEnvironment(catalog),
                 keepColumns, null, keepHidden, ColumnSelectingTransformer.Defaults.IgnoreMissing);
+
+        /// <summary>
+        /// ColumnSelectingEstimator is used to select a list of columns that user wants to keep from a given input.
+        /// </summary>
+        /// <remarks>
+        /// <format type="text/markdown">
+        /// <see cref="SelectColumns(TransformsCatalog, string[])"/> operates on the schema of an input IDataView, dropping unselected columns from the schema.
+        /// </format>
+        /// </remarks>
+        /// <param name="catalog">The transform's catalog.</param>
+        /// <param name="keepColumns">The array of column names to keep.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[SelectColumns](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/SelectColumns.cs)]
```

Comment:
>[!code-csharp[SelectColumns](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/SelectColumns.cs)] [](start = 12, length = 118)

Hi @rogancarr, just curious, do you have some sort of plugin or something that is helping you edit these without error? VS at least with its builtin facilities is giving me very little help here for my own work, just wondering how you are managing.

Reply:
There are actually no debugging tools for these links at the moment. Just very careful coping and pasting and editing.

---
In reply to: [254457544](https://github.com/dotnet/machinelearning/pull/2439#discussion_r254457544) [](ancestors = 254457544)